### PR TITLE
task(rest): drop tweetnacl dependency and unused crypto wrappers

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: client/rest/src/catapult-sdk/crypto/sha3Hasher.js [hunk 1]
```diff
@@ -1,78 +0,0 @@
-/*
```
Comment (gimre-xymcity) [inline client/rest/src/catapult-sdk/crypto/sha3Hasher.js:1]: just to make sure "createHasher part is actually keccak not sha3, right?
(doesn't matter for review, just asking to make sure)

Comment (Jaguar0625) [reply client/rest/src/catapult-sdk/crypto/sha3Hasher.js:1]: no, it was sha3. initially, i think, it was keccak or sha3 based on "configuration".

Comment (gimre-xymcity) [reply client/rest/src/catapult-sdk/crypto/sha3Hasher.js:1]: it _looks_ like keccak (see custom finalize)

but tbh, more worrying part is something similar in code that does exactly that via sha3().update()...
`sha3_256.update(catapult.utils.convert.hexToUint8(encodedPath + branchLinks.join('')))`

this is _probably_ wrong, isn't it?

Comment (Jaguar0625) [reply client/rest/src/catapult-sdk/crypto/sha3Hasher.js:1]: no call to output function { arrayBuffer, hex } ... is finalizing hash (i.e. adding sha3 padding)

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**